src/main.rs

```diff
@@ -155,40 +166,74 @@ fn main() {
         .map(fs::read_to_string)
         .map(|x| x.expect("Unable to open or read from the configuration file"));
 
+    let no_api = cmd_arguments.is_present("no-api");
+
     let shared_info = Arc::new(RwLock::new(InstanceInfo {
         state: InstanceState::Uninitialized,
         id: instance_id,
         vmm_version: crate_version!().to_string(),
     }));
-    let mmds_info = MMDS.clone();
-    let (to_vmm, from_api) = channel();
-    let server =
-        ApiServer::new(mmds_info, shared_info.clone(), to_vmm).expect("Cannot create API server");
-
-    let api_event_fd = server
-        .get_event_fd_clone()
-        .expect("Cannot clone API eventFD.");
-
-    let _vmm_thread_handle = vmm::start_vmm_thread(
-        shared_info,
-        api_event_fd,
-        from_api,
-        seccomp_level,
-        vmm_config_json,
-    );
-
-    match server.bind_and_run(bind_path, start_time_us, start_time_cpu_us, seccomp_level) {
-        Ok(_) => (),
-        Err(Error::Io(inner)) => match inner.kind() {
-            ErrorKind::AddrInUse => panic!("Failed to open the API socket: {:?}", Error::Io(inner)),
-            _ => panic!(
-                "Failed to communicate with the API socket: {:?}",
-                Error::Io(inner)
-            ),
-        },
-        Err(eventfd_err @ Error::Eventfd(_)) => {
-            panic!("Failed to open the API socket: {:?}", eventfd_err)
-        }
+
+    if no_api {
+        let (_, from_api) = channel();
+        let event_fd = EventFd::new()
+            .map_err(Error::Eventfd)
+            .expect("'cannot create dummy Eventfd.");
+
+        vmm::start_vmm(
+            shared_info,
+            event_fd,
+            from_api,
+            seccomp_level,
+            vmm_config_json,
+        );
+    } else {
+        let (to_vmm, from_api) = channel();
+        let mmds_info = MMDS.clone();
+
+        let event_fd = EventFd::new()
+            .map_err(Error::Eventfd)
+            .expect("'cannot create Eventfd.");
+
+        let api_event_fd = event_fd.try_clone().expect("cannot clone Eventfd.");
+        let api_shared_info = shared_info.clone();
+
+        thread::Builder::new()
+            .name("fc_api".to_string())
```

Comment:
nit: I prefer to use to_owned() instead of to_string() in this case.

Reply:
Done